scripts/stake-apecoin.ts

```diff
@@ -0,0 +1,527 @@
+    await hre.network.provider.send("evm_increaseTime", [18000]);
+    await staking.updatePool(1);
+    await staking.updatePool(2);
+    await staking.updatePool(3);
+
+    const pr1 = await staking.pendingRewards(1, av1.address, 3518);
+    const pr2 = await staking.pendingRewards(2, av2.address, 11706);
+    const pr3 = await staking.pendingRewards(3, av3.address, 6037);
+    const pr4 = await staking.pendingRewards(3, av4.address, 3292);
+
+    console.log("Pending Rewards User 1:", pr1.toString());
+    console.log("Pending Rewards User 2:", pr2.toString());
+    console.log("Pending Rewards User 3:", pr3.toString());
+    console.log("Pending Rewards User 4:", pr4.toString());
+
+    cd1 = staking.interface.encodeFunctionData("claimSelfBAYC", [[3518]]);
+    cd2 = staking.interface.encodeFunctionData("claimSelfMAYC", [[11706]]);
+    cd3 = staking.interface.encodeFunctionData("claimSelfBAKC", [
+        [{ mainTokenId: 1044, bakcTokenId: 6037 }],
+        [],
+    ]);
+    cd4 = staking.interface.encodeFunctionData("withdrawBAKC", [
+        [],
+        [{ mainTokenId: 21026, bakcTokenId: 3292, amount: apeAmount }]
+    ]);
+
+    // Have users 1, 2, 3 claim, have user 4 withdraw
+    const balanceBefore1 = await ape.balanceOf(av1.address);
+    await av1.connect(user1).call(staking.address, cd1);
+    const balanceAfter1 = await ape.balanceOf(av1.address);
+    const earned1 = balanceAfter1.sub(balanceBefore1);
+
+    const balanceBefore2 = await ape.balanceOf(av2.address);
+    await av2.connect(user2).call(staking.address, cd2);
+    const balanceAfter2 = await ape.balanceOf(av2.address);
+    const earned2 = balanceAfter2.sub(balanceBefore2);
+
+    const balanceBefore3 = await ape.balanceOf(av3.address);
+    await av3.connect(user3).call(staking.address, cd3);
+    const balanceAfter3 = await ape.balanceOf(av3.address);
+    const earned3 = balanceAfter3.sub(balanceBefore3);
+
+    const balanceBefore4 = await ape.balanceOf(av4.address);
+    await av4.connect(user4).call(staking.address, cd4);
+    const balanceAfter4 = await ape.balanceOf(av4.address);
+    const earned4 = balanceAfter4.sub(balanceBefore4);
+
+    console.log("Claimed User 1:", earned1.toString());
+    console.log("Claimed User 2:", earned2.toString());
+    console.log("Claimed User 3:", earned3.toString());
+    console.log("Claimed User 4:", earned4.toString());
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    ///////////////////////////////              STEP 8: END STAKING              /////////////////////////////////////
+    ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+
+    // Finish the program
+    await hre.network.provider.send("evm_increaseTime", [18000]);
+
+    // Repay all loans
+    await token.connect(user1).approve(REPAYMENT_CONTROLLER, apeAmount.mul(2));
```

Comment:
nit, but the apeAmount*2 seemed weird to me here, maybe ethers.utils.parseEther("10.1") to show the exact from terms